src/visualization/Slider.cc

```diff
@@ -0,0 +1,71 @@
+#include "Slider.h"
+
+using namespace std;
+
+Slider::Slider(unsigned int x_pos, unsigned int y_pos, unsigned int min, unsigned int max, unsigned int& variable, 
+               string name){
+    this->x_pos = x_pos;
+    this->y_pos = y_pos;
+    this->min = min;
+    this->max = max;
+    this->movable = false;
+    this->name = name;
+
+    if (variable >= min && variable <= max) {
+        value = variable;
+    } else {
+        value = ceil((min + max) / 2);
+        variable = value;
+    }
+
+    axis_height = 20;
+    axis_width = 600;
+    slider_width = 20;
+
+    if (!font.loadFromFile("Amiko-Regular.ttf"))
+        std::cout << "Can't find the font file" << std::endl;
+
+    text.setFont(font);
+    text.setColor(sf::Color::Black);
+
+    axis.setPosition(x_pos, y_pos);
+    axis.setFillColor(sf::Color(100, 100, 100));
+    axis.setOutlineColor(sf::Color::Black);
+    axis.setOutlineThickness(1);
+    axis.setSize(sf::Vector2f(axis_width, axis_height));
+    axis.setOrigin(0, axis_height / 2);
+
+    slider_bar.setPosition(x_pos + (axis_width / (max - min)) * (value - min), y_pos);
+    slider_bar.setFillColor(sf::Color(200, 200, 200));
+    slider_bar.setOutlineColor(sf::Color::Black);
+    slider_bar.setOutlineThickness(1);
+    slider_bar.setSize(sf::Vector2f(slider_width, axis_height * 3));
+    slider_bar.setOrigin(slider_width / 2, axis_height * 1.5);
+}
+
+void Slider::draw(sf::RenderWindow& window, unsigned int& number){
+    auto mouse_position = window.mapPixelToCoords(sf::Mouse::getPosition(window));
+    if (sf::Mouse::isButtonPressed(sf::Mouse::Left) && slider_bar.getGlobalBounds().contains(mouse_position))
+        movable = true;
+    if (!sf::Mouse::isButtonPressed(sf::Mouse::Left))
+        movable = false;
+    if (movable && x_pos <= mouse_position.x && mouse_position.x <= x_pos + axis_width) {
```

Comment:
It can be is very tedious to set the minimum or maximum length of sequences.  A possible fix could be to allow mouse positions a little past the end.